tough/src/lib.rs

```diff
@@ -266,28 +271,212 @@ impl<'a, T: Transport> Repository<'a, T> {
+            path: targets_outdir.as_ref(),
+        })?;
+
+        // Fetch targets and save them to the outdir
+        if let Some(target_list) = targets_subset {
+            for target_name in target_list.iter() {
+                self.cache_target(&targets_outdir, target_name)?;
+            }
+        } else {
+            let targets = &self.targets.signed.targets;
+            for target_name in targets.keys() {
+                self.cache_target(&targets_outdir, target_name)?;
+            }
+        }
+
+        // Save the snapshot, targets and timestamp metadata files, and (optionally) the root files.
+        // TODO - support role delegation https://github.com/awslabs/tough/issues/5
+        self.cache_file_from_transport(
+            self.snapshot_filename().as_str(),
+            self.max_snapshot_size()?,
+            "timestamp.json",
+            &metadata_outdir,
+        )?;
+        self.cache_file_from_transport(
+            self.targets_filename().as_str(),
+            self.limits.max_targets_size,
+            "max_targets_size argument",
+            &metadata_outdir,
+        )?;
+        self.cache_file_from_transport(
+            self.timestamp_filename().as_str(),
+            self.limits.max_timestamp_size,
+            "max_timestamp_size argument",
+            &metadata_outdir,
+        )?;
+        if cache_root_chain {
+            // Copy all versions of root.json less than or equal to the current version.
+            for ver in (1..=self.root.signed.version.get()).rev() {
+                let root_json_filename = format!("{}.root.json", ver);
+                self.cache_file_from_transport(
+                    root_json_filename.as_str(),
+                    self.limits.max_root_size,
+                    "max_root_size argument",
+                    &metadata_outdir,
+                )?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Prepends the version number to the snapshot.json filename if using consistent snapshot mode.
+    fn snapshot_filename(&self) -> String {
+        if self.root.signed.consistent_snapshot {
+            format!("{}.snapshot.json", self.snapshot.signed.version)
+        } else {
+            "snapshot.json".to_owned()
+        }
+    }
+
+    /// Prepends the version number to the targets.json filename if using consistent snapshot mode.
+    fn targets_filename(&self) -> String {
+        if self.root.signed.consistent_snapshot {
+            format!("{}.targets.json", self.targets.signed.version)
+        } else {
+            "targets.json".to_owned()
+        }
+    }
+
+    /// Prepends the version number to the timestamp.json filename if using consistent snapshot mode.
+    fn timestamp_filename(&self) -> String {
+        if self.root.signed.consistent_snapshot {
+            format!("{}.timestamp.json", self.snapshot.signed.version)
+        } else {
+            "timestamp.json".to_owned()
+        }
+    }
```

Comment:
This isn't correct -- timestamp.json doesn't have the version prepended even when consistent_snapshot is true.